src/SourceBuild/content/repo-projects/Directory.Build.props

```diff
@@ -114,6 +115,8 @@
 
     <!-- https://github.com/dotnet/source-build/issues/3081 -->
     <EnvironmentVariables Include="CheckEolTargetFramework=false" />
+
+    <EnvironmentVariables Include="FullAssemblySigningSupported=$(FullAssemblySigningSupported)" />
```

Comment:
One disadvantage of this approach is that it introduces a difference in the way the full product source-build behaves versus the individual repos when built in the source-build configuration.  Have you considered moving this into the ArPow infra in arcade?

Reply:
With a change like this in arcade, would it be much harder for someone to turn on Full Signing? IIRC, we were talking about how modifying inner args is much harder from the VMR level? 

```diff
diff --git a/src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/SourceBuildArcadeBuild.targets b/src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/SourceBuildArcadeBuild.targets
index 3933e436..da02d040 100644
--- a/src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/SourceBuildArcadeBuild.targets
+++ b/src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/SourceBuildArcadeBuild.targets
@@ -13,6 +13,8 @@
     <InnerSourceBuildRepoRoot Condition="'$(InnerSourceBuildRepoRoot)' == ''">$(CurrentRepoSourceBuildSourceDir)</InnerSourceBuildRepoRoot>
 
     <CleanInnerSourceBuildRepoRoot Condition="'$(CleanInnerSourceBuildRepoRoot)' == ''">true</CleanInnerSourceBuildRepoRoot>
+
+    <FullAssemblySigningSupported Condition="'$(FullAssemblySigningSupported)' != ''">false</FullAssemblySigningSupported>
   </PropertyGroup>
 
   <Target Name="ExecuteWithSourceBuiltTooling"
@@ -87,6 +89,8 @@
 
       <InnerBuildArgs>$(InnerBuildArgs) /p:DotNetBuildOffline=true</InnerBuildArgs>
       <InnerBuildArgs Condition=" '$(DotNetPackageVersionPropsPath)' != '' ">$(InnerBuildArgs) /p:DotNetPackageVersionPropsPath=$(DotNetPackageVersionPropsPath)</InnerBuildArgs>
+
+      <InnerBuildArgs>$(InnerBuildArgs) /p:FullAssemblySigningSupported=$(FullAssemblySigningSupported)</InnerBuildArgs>
     </PropertyGroup>
 
     <ItemGroup>
```

Reply:
> would it be much harder for someone to turn on Full Signing? 

I don't think so, you could still override via an EnvironmentVariable.

> IIRC, we were talking about how modifying inner args is much harder from the VMR level?

The pieces that make it _hard_ don't apply here IMO.  Flowing inner args gets complicated because because there are multiple use cases to consider - e.g. does the arg apply to only the outer build, inner build, both outer and inner.